### DISCUSSION TITLE
operator sharding and pipelining support in horovod

### DISCUSSION THREAD

**Discussion by DavidPeleg6** [horovod/horovod#2894] [Q&A]
I am currently trying to split a model that fits on multiple gpu's and I was wondering whether there was a more generic way to do so with horovod rather than tf.distribute .
More specifically, I would like to implement something similar to PipeDream but also have support for splitting Dense layers across multiple gpu's. Also, I would like to explore cases where specific layers receive a split batch while others get the full one.
I came across AliBaba's Whale framework that claims to do this exact task but it seems that the API is not publicly available.

Is there a way to implement this in Horovod?

**Comment by maxhgerlach** [comment]
> support for splitting Dense layers across multiple gpu's

To implement model parallelism of the Megatron-LM variety with Horovod, the feature discussed in issue #2139 will be helpful. At this time there are two preliminary pull requests aiming to provide that, which will hopefully converge in a bit (I have been working on one of them).
